tools/mgba-rom-test-hydra/main.c

```diff
@@ -101,7 +101,7 @@ static void handle_read(struct Runner *runner)
 add_to_results:
                     runner->results++;
                     soc += 2;
-                    fprintf(stdout, "%s: ", runner->test_name);
+                    fprintf(stdout, "[%d] %s: ", runner->pid, runner->test_name);
```

Comment:
I don't think you intended to commit this change. (It was used to work out which process was running which tests, but I think the changes to improve the "my tests fail when run together, but not when run alone" experience should be done on a different PR)

Reply:
I left this in intentionally as it's quite useful and very trivial. If you really think I should remove it, I'll do so but I don't really see the need.

Reply:
I like printing something before the tests, but I think it'd be better if it was the test runner number (which is stable) rather than the PID (which isn't). If it was the runner number we could imagine being able to re-run the test suite in a mode that's like "run the tests which would be run by runner number X", and if we also inherited `make`'s `-j` parameter then we would be able to easily reproduce these sorts of errors on another developer's machine.

We definitely have a bad story for debugging these sorts of interferences between test cases, so if you want to take a look at doing a PR that improves the whole experience that would be great—if not, no worries I'll get to it eventually :)